src/openssl/hmac.cr

```diff
@@ -19,7 +21,7 @@ class OpenSSL::HMAC
     key_slice = key.to_slice
     data_slice = data.to_slice
     buffer = Slice(UInt8).new(128)
-    LibCrypto.hmac(evp, key_slice, key_slice.size, data_slice, data_slice.size, buffer, out buffer_len)
+    LibCrypto.hmac(evp.not_nil!, key_slice, key_slice.size, data_slice, data_slice.size, buffer, out buffer_len)
```

Comment:
It shouldn't be, but I was getting a compiler error after adding the conditional macro code on line 11. I'll investigate further to confirm that this is a reproducable problem, if so it may indicate a bug in the compiler.